elixir/apps/web/lib/web/auth.ex

```diff
@@ -490,6 +507,21 @@ defmodule Web.Auth do
     %{}
   end
 
+  def put_client_auth_state_from_cookie(%Plug.Conn{} = conn, _) do
+    conn = Plug.Conn.fetch_cookies(conn, signed: [@client_auth_cookie_name])
+
+    case cookie = conn.cookies[@client_auth_cookie_name] do
+      [actor_name: _, fragment: _, identity_provider_identifier: _, state: _] ->
+        Plug.Conn.assign(conn, :client_auth_data, cookie)
+
+      _ ->
+        conn
+        |> Phoenix.Controller.put_flash(:error, "Client authentication failed. Please try again.")
+        |> Phoenix.Controller.redirect(to: ~p"/#{conn.path_params["account_id_or_slug"]}")
```

Comment:
This redirect will break the sign-in cycle for the client because you won't be able to do a 2nd attempt of a client sign-in because there is no `as` and other required params anymore. So even if you are successful 2nd time you will see the portal dashboard in a webview as a result..

So there are two solutions to this:
1. Whenever possible, keep the params and use them in a redirect;
2. Otherwise, show an error page and ask user to close the webview and try to sign in again (manually reset the state).

Reply:
Yeah, the main reason I put that in was as a guard against someone trying to just go directly to `/sign_in/deep_link`, in which case they'd have no params to use.  With that in mind, I think I'll just show an error page as you suggested, rather than trying to handle the two failure cases separately.

Reply:
Updated.